golang/wasm/wrapper_js.go

```diff
@@ -436,40 +437,89 @@ func newUpdater(_ js.Value, args []js.Value) any {
 	Returns a Promise which executes a DOH query using sig0.DefaultDOHResolver
 
 arguments:
-- 0: (required) name to query
-- 1: (optional) dns RRtype, defaults to A
+  - 0: (required) - either:
+  - string: domain name to query
+  - object: {domain, type, dohurl}
+  - 1: (optional) - either
+  - string: dns RRtype, defaults to A
+  - object: {type, dohurl}
 */
 func query(_ js.Value, args []js.Value) any {
 	handler := js.FuncOf(func(this js.Value, promises []js.Value) interface{} {
 		resolve := promises[0]
 		reject := promises[1]
 
-		var rrType uint16 = dns.TypeA
+		var (
+			domainName  string
+			rrType      uint16 = dns.TypeA
+			err         error
+			dohResolver = sig0.DefaultDOHResolver
+		)
+
+		decodeOpts := func(opt js.Value) error {
+			if dn := opt.Get("domain").String(); dn != "" {
+				domainName = dn
+			}
+			if optType := opt.Get("type"); optType.Type() == js.TypeString {
+				if rrString := optType.String(); rrString != "" {
+					rrType, err = sig0.QueryTypeFromString(rrString)
+					if err != nil {
+						return err
+					}
+				}
+			}
+			if du := opt.Get("dohurl"); du.Type() == js.TypeString {
+				duStr := du.String()
+				duUrl, err := url.Parse(duStr)
+				if err == nil {
+					dohResolver = duUrl.Host
```

Comment:
`dohurl` also supports receiving a full url but actually only takes the host from that. we would need to patch `github.com/shynome/doh-client` for it to construct it's url differently here https://github.com/shynome/doh-client/blob/3b78e35dadc53df5cf253bde3692e59507584a92/conn.go#L30

Reply:
It's good that dohurl supports the full URL, as the JS function findDOHEndpoint() in pure.js in its current form picks up the dohUrl as a single URL string, not as components. Javascipt should only query our SVCB RR entries for each enabled zone which will work as long as the defacto example path in RFC8484 is used. Issue raised in doh-client project [here](https://github.com/shynome/doh-client/issues/1).